From a166d5450e3f47587b98f6005f9f5627dbe21a5b Mon Sep 17 00:00:00 2001
From: Akim Demaille <akim.demaille@gmail.com>
Date: Tue, 2 Aug 2022 08:45:25 +0200
Subject: [PATCH] yacc.c, lalr1.cc: fix warnings about yynerrs

Most often yynerrs is not used.  Mark it unused so parsers generated by Bison
3.8.2 continue to compile under toolchains which promote that warning to an
error.  Upstream commit a166d5450e3f47587b98f6005f9f5627dbe21a5b.

diff --git a/data/skeletons/lalr1.cc b/data/skeletons/lalr1.cc
index 240f4c362..4242b992d 100644
--- a/data/skeletons/lalr1.cc
+++ b/data/skeletons/lalr1.cc
@@ -847,7 +847,7 @@ m4_if(b4_prefix, [yy], [],
     int yylen = 0;
 
     // Error handling.
-    int yynerrs_ = 0;
+    int yynerrs_ YY_ATTRIBUTE_UNUSED = 0;
     int yyerrstatus_ = 0;
 
     /// The lookahead symbol.
diff --git a/data/skeletons/yacc.c b/data/skeletons/yacc.c
index fd7e2c33f..07a523b0a 100644
--- a/data/skeletons/yacc.c
+++ b/data/skeletons/yacc.c
@@ -256,7 +256,7 @@ int yynerrs;]])])
 m4_define([b4_declare_parser_state_variables],
 [b4_pure_if([[
     /* Number of syntax errors so far.  */
-    int yynerrs]m4_ifval([$1], [ = 0])[;
+    int yynerrs YY_ATTRIBUTE_UNUSED]m4_ifval([$1], [ = 0])[;
 ]])[
     yy_state_fast_t yystate]m4_ifval([$1], [ = 0])[;
     /* Number of tokens to shift before error messages enabled.  */
